From a65af40dd5f3c5711e4790e0b53a89bd90cac71a Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@intel.com>
Date: Fri, 30 Aug 2013 13:53:05 +0300
Subject: SQUISH [Klockwork] Fix null pointer reference in NetworkController

This patch should be squashed with b414a3e (Add new Ethernet network
type) and never upstreamed as a separate patch.

Category: Device Enablement
Domain: Connectivity/Ethernet
Upstream-Candidate: no

Change-Id: I63b44da8cd1bc2cfd274fce11375efd99e287aec
For: AXIA-3946
Signed-off-by: Robert Chiras <robert.chiras@intel.com>
---
 .../statusbar/policy/NetworkController.java        |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
index 4a42649..be99a00 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkController.java
@@ -968,8 +968,10 @@ public class NetworkController extends BroadcastReceiver implements DemoMode {
             mEthernetEnabled = true;
             final EthernetInfo ei = (EthernetInfo)
                     intent.getParcelableExtra(EthernetManager.EXTRA_ETHERNET_INFO);
-            final NetworkInfo networkInfo = ei.getNetworkInfo();
-            mEthernetConnected = networkInfo != null && networkInfo.isConnected();
+            if (ei != null) {
+                final NetworkInfo networkInfo = ei.getNetworkInfo();
+                mEthernetConnected = networkInfo != null && networkInfo.isConnected();
+            }
         }
         updateEthernetIcons();
     }
-- 
1.7.4.1

